stream.go

```diff
@@ -158,21 +119,151 @@ func (s *Stream) MeasurementLatest(p Params) (<-chan *measurement.Result, error)
     return ch, nil
 }
 
+// MeasurementLatestWithChan streams the latest measurement results, as described
+// by the Params, and sends them to the supplied channel. The channel supplied should
+// not be closed, but should be reused between connections of the same measurement.
+// A channel is returned that will receive an empty struct and be closed when the connection has been closed.
+//
+// Params available are:
+//
+// "msm": int - The measurement id to get results from.
+//
+// "type": string - The measurement result type to stream.
+//
+// "sourceAddress": none - Unimplemented
+//
+// "sourcePrefix": none - Unimplemented
+//
+// "destinationAddress": none - Unimplemented
+//
+// "destinationPrefix": none - Unimplemented
+//
+// "passThroughHost": none - Unimplemented
+//
+// "passThroughPrefix": none - Unimplemented
+//
+// "sendBacklog": none - Unimplemented
+//
+// "buffering": none - Unimplemented
+func (s *Stream) MeasurementLatestWithChan(p Params, ch chan *measurement.Result) (<-chan struct{}, error) {
+    c, subscribe, err := subscribeAndDial(p)
+    if err != nil {
+        return nil, err
+    }
+
+    closedChan := make(chan struct{})
+
+    err = c.On("atlas_error", func(h *gosocketio.Channel, args interface{}) {
+        r := &measurement.Result{ParseError: fmt.Errorf("atlas_error: %v", args)}
+        ch <- r
+        c.Close()
+    })
+    if err != nil {
+        return closedChan, fmt.Errorf("c.On(atlas_error): %s", err.Error())
```

Comment:
Not sure if I should return `nil, fmt.Errorf...` here instead. I don't know what effect an error on the `c.On()` method has on the connection